sdk/python/feast/offline_server.py

```diff
@@ -0,0 +1,323 @@
+
+        Returns:
+            Returns either the specified feature view, or raises an exception if
+            none is found
+        """
+        try:
+            fv = self.store.registry.get_feature_view(name=fv_name, project=project)
+            if name_alias is not None:
+                for fs in self.store.registry.list_feature_services(project=project):
+                    for p in fs.feature_view_projections:
+                        if p.name_alias == name_alias:
+                            logger.debug(
+                                f"Found matching FeatureService {fs.name} with projection {p}"
+                            )
+                            fv = fv.with_projection(p)
+            return fv
+        except Exception:
+            try:
+                return self.store.registry.get_stream_feature_view(
+                    name=fv_name, project=project
+                )
+            except Exception as e:
+                logger.error(
+                    f"Cannot find any FeatureView by name {fv_name} in project {project}"
+                )
+                raise e
+
+    def list_feature_views_by_name(
+        self, feature_view_names: List[str], name_aliases: List[str], project: str
+    ) -> List[FeatureView]:
+        return [
+            remove_dummies(
+                self.get_feature_view_by_name(
+                    fv_name=fv_name, name_alias=name_aliases[index], project=project
+                )
+            )
+            for index, fv_name in enumerate(feature_view_names)
+        ]
+
+    # Extracts the API parameters from the flights dictionary, delegates the execution to the FeatureStore instance
+    # and returns the stream of data
+    def do_get(self, context, ticket):
+        key = ast.literal_eval(ticket.ticket.decode())
+        if key not in self.flights:
+            logger.error(f"Unknown key {key}")
+            return None
+
+        command = json.loads(key[1])
+        api = command["api"]
+        logger.debug(f"get command is {command}")
+        logger.debug(f"requested api is {api}")
+        try:
+            if api == OfflineServer.get_historical_features.__name__:
+                table = self.get_historical_features(command, key).to_arrow()
+            elif api == OfflineServer.pull_all_from_table_or_query.__name__:
+                table = self.pull_all_from_table_or_query(command).to_arrow()
+            elif api == OfflineServer.pull_latest_from_table_or_query.__name__:
+                table = self.pull_latest_from_table_or_query(command).to_arrow()
+            else:
+                raise NotImplementedError
+        except Exception as e:
+            logger.exception(e)
+            traceback.print_exc()
+            raise e
+
+        # Get service is consumed, so we clear the corresponding flight and data
+        del self.flights[key]
+        return fl.RecordBatchStream(table)
+
+    def offline_write_batch(self, command, key):
```

Comment:
it is not a mandatory to give the types, but it makes it more readable to have types unless it is expected to have any data type.

Reply:
You mean in the methods parameters or the return value or both?
I guess you mean for all the methods and not only this specific one?

Reply:
I was specifically about method parameters. but if there is no change in functionality better to add types to return variable also.

Reply:
@lokeshrangineni the change implemented under [commit](https://github.com/feast-dev/feast/pull/4262/commits/bdf0150bd00608079122f41a4ebb71bb432f4ee3) thanks @tmihalac addressing it